pkg/services/provisioning/dashboards/file_reader.go

```diff
@@ -412,42 +413,139 @@ type dashboardIdentity struct {
 }
 
 func (d *dashboardIdentity) Exists() bool {
-	return len(d.title) > 0 && d.folderID > 0
+	return len(d.title) > 0
 }
 
-func newProvisioningSanityChecker(provisioningProvider string) provisioningSanityChecker {
-	return provisioningSanityChecker{
-		provisioningProvider: provisioningProvider,
-		uidUsage:             map[string]uint8{},
-		titleUsage:           map[dashboardIdentity]uint8{},
+func newUsageTracker() *usageTracker {
+	return &usageTracker{
+		uidUsage:   map[string]uint8{},
+		titleUsage: map[dashboardIdentity]uint8{},
 	}
 }
 
-type provisioningSanityChecker struct {
-	provisioningProvider string
-	uidUsage             map[string]uint8
-	titleUsage           map[dashboardIdentity]uint8
+type usageTracker struct {
+	uidUsage   map[string]uint8
+	titleUsage map[dashboardIdentity]uint8
 }
 
-func (checker provisioningSanityChecker) track(pm provisioningMetadata) {
+func (t *usageTracker) track(pm provisioningMetadata) {
 	if len(pm.uid) > 0 {
-		checker.uidUsage[pm.uid]++
+		t.uidUsage[pm.uid]++
 	}
 	if pm.identity.Exists() {
-		checker.titleUsage[pm.identity]++
+		t.titleUsage[pm.identity]++
 	}
 }
 
-func (checker provisioningSanityChecker) logWarnings(log log.Logger) {
-	for uid, times := range checker.uidUsage {
-		if times > 1 {
-			log.Error("the same 'uid' is used more than once", "uid", uid, "provider", checker.provisioningProvider)
+type set struct {
+	data map[string]struct{}
+}
+
+func newSet() *set {
+	return &set{make(map[string]struct{})}
+}
+
+func (s *set) Add(entry string) {
+	s.data[entry] = struct{}{}
+}
+
+func (s *set) AsSlice() []string {
+	entries := make([]string, len(s.data))
+	index := 0
+
+	for entry := range s.data {
+		entries[index] = entry
+		index++
+	}
+
+	return entries
+}
+
+type duplicate struct {
+	Readers *set
+	Sum     uint8
+}
+
+func newDuplicate() *duplicate {
+	return &duplicate{Readers: newSet()}
+}
+
+type duplicateEntries struct {
+	Title map[dashboardIdentity]*duplicate
+	UID   map[string]*duplicate
```

Comment:
Why are these fields not named in plural, since they are collections?
```suggestion
	Titles map[dashboardIdentity]*duplicate
	UIDs   map[string]*duplicate
```

Reply:
My bad, I will fix it.

Reply:
Fixed.